### PR TITLE
gh-113407: Fix import of unittest.mock when CPython is built without docstrings

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-113407

### CODE REVIEW

File: Lib/unittest/mock.py [hunk 1]
```diff
@@ -2250,9 +2253,12 @@ def __init__(self, /, *args, **kwargs):
         self.__dict__['_mock_await_count'] = 0
         self.__dict__['_mock_await_args'] = None
         self.__dict__['_mock_await_args_list'] = _CallList()
-        code_mock = NonCallableMock(spec_set=_CODE_ATTRS)
-        code_mock.__dict__["_spec_class"] = CodeType
-        code_mock.__dict__["_spec_signature"] = _CODE_SIG
+        if _CODE_SIG:
+            code_mock = NonCallableMock(spec_set=_CODE_ATTRS)
+            code_mock.__dict__["_spec_class"] = CodeType
+            code_mock.__dict__["_spec_signature"] = _CODE_SIG
```
Comment (cjw296) [inline Lib/unittest/mock.py:2259]: What functionality is lost by having `_spec_class` and `_spec_signature` missing here?
These were evidently there for a reason, but if no tests are failing without them that suggest they might be superfluous and could be dropped completely, or, more likely, that there's tests coverage that needs adding.

The backport has 100% line coverage checking as part of its CI, so I'm a little surprised.

**Review by cjw296 — COMMENTED** [review]
I'm worried about what needs these to be there in the first place breaking.

I'm also curious: why would you want to build a python distribution without docstrings?

File: Lib/unittest/mock.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/unittest/mock.py:2259]: Was not it a pure optimization in #83076? This PR simply disables it if the signature is not available and makes the code to go the old way.

**Comment by serhiy-storchaka** [comment]
Building a python distribution without docstrings is a supported feature. Currently the mock module is broken in this configuration, and it affects many tests.

File: Lib/unittest/mock.py (hunk 1 above)
Comment (cjw296) [reply Lib/unittest/mock.py:2259]: Yeah, I guess if there's no signature available then there's no code signature available.
Where do we go to see the test suite running (and passing after this PR!) with docstrings disabled?

**Comment by cjw296** [comment]
> Building a python distribution without docstrings is a supported feature

I didn't say it wasn't, I asked why would you want to build a python distribution without docstrings.
I'm still curious :-)

**Review by cjw296 — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
I have no other reasons except supporting Python bug-free in all configurations. We do no have buildbots for all possible combinations of configuration options, so we depend on manual testing here. Time from time we test and fix builds in nonstandard configurations, but last time it was years ago. There was an old issue #81682 with an outdated PR, and it is the reason why I do this right now.

**Pull request merged**